[LeeZeHao] iP

LeeZeHao opened the pull request:
# Stelle

> "When there is the chance to make a choice, make one that you know you won't regret..." - Stelle (Honkai: Star Rail)

Welcome to the Astral Express! Stelle will help you remember what you needed to do.

The interface is:
- Text based
- Lightning fast
- Super quick to learn

### Installation
1. Download it from [here](https://github.com/LeeZeHao/ip/releases/tag/v0.1.0).
2. Double click to run.
3. Add your tasks.
4. Enjoy better task management! 😊 

Completely **FREE**

### Features
- [X] Managing tasks
- [X] Managing to-dos
- [X] Managing deadlines
- [x] Managing events
- [ ] Reminders (coming soon!) 

----
Here's the `main` method:
```java
package stelle;

/** Represents the main chatbot class.
 * @author Lee Ze Hao (A0276123J)
 */

public class Stelle {
    static final String NAME = "stelle";
    static final String FILE_PATH = "./data/stelle.txt";

    private Ui ui;

    public Stelle(String name, String filePath) {
        ui = new Ui(name, filePath);
    }

    public void run() {
        ui.run();
    }

    public static void main(String[] args) {
        new Stelle(NAME, FILE_PATH).run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

LeeZeHao committed:
Change Duke.java to Stelle.java and add Greet and Exit

LeeZeHao committed:
Refactor chat logic to use ChatLogic.java

LeeZeHao committed:
Add echo user input and bye command

LeeZeHao committed:
ChatLogic.java: Add static constants for commands

LeeZeHao committed:
Add Task.java

LeeZeHao committed:
ChatLogic.java: Add list and add task features

LeeZeHao committed:
Task.java: Add "done" status to task

LeeZeHao committed:
Add mark and unmark task features

LeeZeHao committed:
Add Deadline.java, Event.java, ToDo.java

LeeZeHao committed:
Add time functions to Deadline.java, Event,java

LeeZeHao committed:
Add creation of ToDo, Deadline, Event type tasks

LeeZeHao committed:
Setup automated test script for previous levels

LeeZeHao committed:
Create DukeException.java

LeeZeHao committed:
Add exception handling for empty todo desc and wrong command

LeeZeHao committed:
Add deadline and event empty desc exceptions

LeeZeHao committed:
Add NoSuchTaskException for non-existent task numbers

LeeZeHao committed:
Add NoSuchTaskException for non-existent task numbers

LeeZeHao committed:
ChatLogic.java: Refactor to use ArrayList for task list

LeeZeHao committed:
Add mark and deletion task number not specified exception

LeeZeHao committed:
Complete task deletion function

LeeZeHao committed:
updated automatic TextUiTesting to include deletion

LeeZeHao committed:
Add javadoc comments for all classes

LeeZeHao committed:
Add Storage.java

LeeZeHao committed:
Add methods to Task objects for reading of more internal data

LeeZeHao committed:
Storage.java: Add writeTasksToFile method body

LeeZeHao committed:
Add TaskWithoutTypeException.java

LeeZeHao committed:
Storage.java: Add readTasksFromFile method body

LeeZeHao committed:
Implement Storage functions into ChatLogic class

LeeZeHao committed:
bug fix: Fix not creating new file if needed on startup

LeeZeHao committed:
Merge branch 'branch-Level-7'

LeeZeHao committed:
Add WrongDateFormatException.java

LeeZeHao committed:
Add LocalDateTime support to Task classes

LeeZeHao committed:
bug fix: Fix various input, output and saving related bugs

LeeZeHao committed:
Merge branch 'branch-Level-8'

LeeZeHao committed:
Add StorageWriteException.java

LeeZeHao committed:
Add TaskList.java

LeeZeHao committed:
Integrated TaskList class with existing logic

LeeZeHao committed:
Add Ui.java

LeeZeHao committed:
Rename ChatLogic class to Parser

LeeZeHao committed:
Refactor Ui.java to encompass user interaction more

LeeZeHao committed:
Merge branch 'branch-A-MoreOOP'

LeeZeHao committed:
Organize all classes into suitable java packages

LeeZeHao committed:
Merge branch 'branch-A-Packages'

LeeZeHao committed:
Merge branch 'add-gradle-support' of https://github.com/LeeZeHao/ip

LeeZeHao committed:
Add gradle + test gradle run and build

LeeZeHao committed:
Merge branch 'branch-A-Gradle'

LeeZeHao committed:
Add DeadLineTest.java ToDoTest.java

LeeZeHao committed:
Update DeadLineTest.java ToDoTest.java

LeeZeHao committed:
Merge branch 'branch-A-JUnit'

LeeZeHao committed:
Update build.gradle for packaging as JAR

LeeZeHao committed:
Merge branch 'branch-A-Jar'

LeeZeHao committed:
Add search to TaskList

LeeZeHao committed:
Implement find task to Parser

LeeZeHao committed:
Tweak method name to comply with SE-EDU Java standard

LeeZeHao committed:
Add JavaDoc comments to more methods

LeeZeHao committed:
Merge branch 'branch-A-JavaDoc'

LeeZeHao committed:
Merge branch 'branch-A-CodingStandard'

LeeZeHao committed:
Merge branch 'branch-Level-9'

LeeZeHao committed:
Setup checkstyle + fix test and exception code standards

LeeZeHao committed:
Fix checkstyle code standards issues

LeeZeHao committed:
Merge branch 'branch-A-CheckStyle'

LeeZeHao committed:
Test basic JavaFX setup (tutorial part 1)

LeeZeHao committed:
Add dialog box

LeeZeHao committed:
Refactor Stelle to use new GUI for input/output

LeeZeHao committed:
Add dialog box flip for user/chatbot

LeeZeHao committed:
Refactor to use FXML

LeeZeHao committed:
MainWindow.fxml: Fix anchors

LeeZeHao committed:
Merge branch 'branch-Level-10'

LeeZeHao committed:
Fix CheckStyle warnings on new code

LynetteLeeSiYing commented on src/main/java/stelle/Parser.java:
```diff
@@ -0,0 +1,289 @@
+import stelle.task.Task;
+import stelle.task.ToDo;
+
+/**
+ * This class encapsulates chatbot parsing and logic.
+ * @author Lee Ze Hao (A0276123J)
+ */
+
```
Well done In the organisation of your code, but there are some slight violations on the javadoc conventions. e.g. there should be a line separated before @param and other tags, there should not be a blank line between the comment block and method.

LynetteLeeSiYing commented on src/main/java/stelle/exception/DeadlineNoDescriptionException.java:
Love the way you handled different exception, with different exception class.

LynetteLeeSiYing commented on src/main/java/stelle/task/Task.java:
Could I instantiate Task class here without instantiating any of its subclass, since it is not abstract?

LynetteLeeSiYing reviewed (commented):
LGTM. Just some coding violations on javadoc comment

kuiktzecheng commented on src/main/java/stelle/Parser.java:
```diff
@@ -0,0 +1,289 @@
+    /**
+     * Constructor for a ChatLogic class. Also fetches data from the specified file path upon construction.
+     * @param ui The stelle.Ui used by the chatbot.
+     * @param filePath The file path used for data storage.
```
your JavaDoc comments are not the right style. There should be a blank line between the description and the @param line. The other JavaDoc comments of the project have the same problem too

can see this page for reference : 
https://se-education.org/guides/conventions/java/intermediate.html#comments

T1duS commented on src/main/java/stelle/exception/StelleException.java:
```diff
@@ -0,0 +1,12 @@
+ * @author Lee Ze Hao (A0276123J)
+ */
+
+public class StelleException extends RuntimeException {
```
Good job on adding so many different exceptions to cover different issue the program could face!!

T1duS commented on src/main/java/stelle/task/Deadline.java:
```diff
@@ -0,0 +1,55 @@
+ */
+
+public class Deadline extends Task {
+    private static final String DATE_TIME_INPUT_FORMATTER_PATTERN = "yyyy-MM-dd HH:mm";
```
Could have tried to cover multiple kinds of input formats

T1duS commented on src/main/java/stelle/TaskList.java:
```diff
@@ -0,0 +1,121 @@
+
+        for (int i = 0; i < taskList.size(); i++) {
+            Task task = taskList.get(i);
+            if (task.getTaskName().toLowerCase().contains(query.toLowerCase())) {
```
It might have been better to let "Task" class handle this rather than TaskList.

T1duS reviewed (commented):
Excellent use of modularization and abstraction. Very good quality OOP code without any obvious mistakes.

izruff commented on src/main/java/stelle/task/Event.java:
```diff
@@ -0,0 +1,67 @@
+            this.from = LocalDateTime.parse(from.strip(), dateTimeInputFormatter);
+            this.to = LocalDateTime.parse(to.strip(), dateTimeInputFormatter);
+        } catch (Exception e) {
+            throw new WrongDateFormatException();
```
Should the date format validation be done outside of the `Event` class, e.g. in the `Parser` class instead? To me, this feels more appropriate since the `Parser` should handle all input-related matters. Additionally, validating the dates in the `Parser` saves you from rewriting a few lines of code in the other `Deadline` class.

izruff reviewed (commented):
Overall, LGTM! Your code base is fully functional and has some neat logic and OOP principles, though there are some places where your code could use some refactoring.

LeeZeHao committed:
build.grade: Change main class to stelle.ui.Launcher

In build.gradle, application {mainClass.set()} denotes the main class
to be run upon launching the .jar file.

Before this, it was stelle.Stelle, as the text UI version launches
from this class's main method.

Currently stelle.ui.Launcher contains the correct launcher to launch
the UI. Doing so enables JAR file built by gradle to use the UI.

LeeZeHao committed:
Parser.java: Add Java assert

There are no Java asserts in the code.

This means there are no ways to check for developer code mistakes and
other bugs during runtime.

Adding Java assert to check for program bugs to following methods:
* processDeleteTaskInput
* markTask
* unmarkTask
* listTasks
allows us to check for potential bugs introduced during further
implementation.

LeeZeHao committed:
Parser.java: Refactor old code

The current code has gone through multiple iterations and has built up
some readability and code quality issues.

This should be changed to enable easier development for future
iterations.

Let's refactor some legacy code to improve code quality.
* Parser.java: Remove reference to Ui class as it is no longer in use
* TaskList.java: Decompose Conditional at Line 119
* TaskList.java: Remove field "filePath" as it is not used outside of
constructor
* Ui.java: Remove field "filePath" as it is not used outside of
constructor

These changes improve code quality and readability, reducing confusion
for future developers.

LeeZeHao committed:
Merge pull request #1 from LeeZeHao/branch-A-Assertions

Parser.java: Add Java assert

LeeZeHao committed:
Merge branch 'master' into branch-A-CodeQuality

LeeZeHao committed:
Merge pull request #2 from LeeZeHao/branch-A-CodeQuality

Refactor old code

LeeZeHao committed:
Add .github\workflows\gradle.yml

Currently continuous integration is not setup.

It should be setup to automate workflow management.

Let's add github\workflows\gradle.yml to setup CI with GitHub Actions.

This is an easy way to integrate CI into an existing GitHub project.

LeeZeHao committed:
Add .github\workflows\gradle.yml

Currently continuous integration is not setup.

It should be setup to automate workflow management.

Let's add github\workflows\gradle.yml to setup CI with GitHub Actions.

This is an easy way to integrate CI into an existing GitHub project.

LeeZeHao committed:
Delete old gradle.yml workflow file

LeeZeHao committed:
Add .github\workflows\gradle.yml

Currently continuous integration is not setup.

It should be setup to automate workflow management.

Let's add github\workflows\gradle.yml to setup CI with GitHub Actions.

This is an easy way to integrate CI into an existing GitHub project.

LeeZeHao committed:
Parser.java: Deal with duplicate items

The chatbot allows multiple tasks with the same name to be added.

This causes confusion for the user who may accidentally add multiples
of one task.

Let's modify addToDo, addDeadline, addEvent to not add a task if a task
with the same name already exists.

This will prevent the user from making a copy of an existing task, and
notify them about it as well, reducing potential confusion.

LeeZeHao committed:
Fix could not add task if name was part of existing task name

Last commit uses old TaskList.find method to check if the task name
already exists.

This is wrong as the old names only needed to contain the new name to
be counted as duplicate (e.g. "SoC career fair" contains "career fair")

Let's add "exact" param to TaskList.find to set whether we are looking
for an exact match.

Results in adding new tasks performing as expected (can add as long as
name is not exact duplicate)

LeeZeHao committed:
Merge branch 'branch-BCD-Extension'

LeeZeHao committed:
Stelle.java: Remove unused run() method

LeeZeHao committed:
DialogBox.java: Change font for chatBot

LeeZeHao committed:
Make different chat font for user and bot

LeeZeHao committed:
Add padding to label in Dialog Box

LeeZeHao committed:
Merge branch 'branch-A-BetterGUI'

LeeZeHao committed:
Alter dialog by bot

LeeZeHao committed:
Change bot response to exceptions

LeeZeHao committed:
Merge branch 'branch-A-Personality'

LeeZeHao committed:
Refactor Deadline and Event construction to pass in date obj directly

LeeZeHao committed:
Add exception for if "To" date before "From" date while creating Event

LeeZeHao committed:
Add exception for no /to /from /by in Event or Deadline creation

LeeZeHao committed:
Merge branch 'branch-A-MoreErrorHandling'

LeeZeHao committed:
Update title of JavaFX window

LeeZeHao committed:
.gitignore: Ignore /data folder

LeeZeHao committed:
Add Ui.png

LeeZeHao committed:
Update README.md

LeeZeHao committed:
Update README.md

LeeZeHao committed:
Update README.md

LeeZeHao committed:
Update README.md

LeeZeHao committed:
Update README.md

LeeZeHao committed:
docs/README.md: Add adding tasks to user guide

LeeZeHao committed:
docs/README.md: add rest of user actions

LeeZeHao committed:
Merge branch 'branch-A-UserGuide'

LeeZeHao committed:
Update README.md

LeeZeHao committed:
Remove unused old code

LeeZeHao committed:
Improve JUnit testing for Deadline getByTime() method

LeeZeHao committed:
Improve JUnit test for Deadline constructor

LeeZeHao committed:
Improve JUnit test for ToDo constructor

LeeZeHao committed:
Improve JUnit test for ToDo mark/unmarking

LeeZeHao committed:
Merge branch 'branch-A-MoreTesting'

LeeZeHao committed:
Update README.md
